compiler/rustc_hir_typeck/src/generator_interior/mod.rs

```diff
@@ -571,7 +571,7 @@ fn check_must_not_suspend_ty<'tcx>(
         // FIXME: support adding the attribute to TAITs
         ty::Alias(ty::Opaque, ty::AliasTy { def_id: def, .. }) => {
             let mut has_emitted = false;
-            for &(predicate, _) in fcx.tcx.explicit_item_bounds(def) {
+            for (predicate, _) in fcx.tcx.explicit_item_bounds(def).subst_identity_iter_copied() {
```

Comment:
I see this more as a skip_binder than a subst_identity conceptually. Can we use that instead?

Reply:
Ah yeah... I missed that it's only looking at the inner `def_id`